File: hr--fair-rations--soln.py
#!/bin/python3


def soln(b):
	odds = []
	for i in range(len(b)):
		if (b[i]%2 != 0):
			odds.append(i)
	if (len(odds)%2 != 0):
		return "NO"
	swaps = 0
	for i in range(1,len(odds),2):
		swaps += odds[i] - odds[i-1]
	return 2*swaps


n = int(input().strip())
b = [int(i) for i in input().strip().split(' ')]
print(soln(b))

File: hr--fair-rations--soln.cpp
#include <iostream>
#include <vector>

using namespace std;

int soln(vector<int> b) {
	vector<int> odds;
	for (int i=0; i<b.size(); i++) {
		if (b[i]%2 != 0) {
			odds.push_back(i);
		}
	}
	if (odds.size()%2 != 0) {
		return -1;
	}
	int swaps = 0;
	for (int i=1; i<odds.size(); i +=2) {
		swaps += odds[i] - odds[i-1];
	}
	return 2*swaps;
}

int main() {
	int n;
	cin >> n;
	vector<int> b(n);
	for (int i=0; i<n; i++) {
		cin >> b[i];
	}
	int s = soln(b);
	if (s == -1) {
		cout << "NO" << endl;
	}
	else {
		cout << s << endl;
	}
	return 0;
}
